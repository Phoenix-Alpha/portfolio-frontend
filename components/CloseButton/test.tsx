import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { CloseButton } from './index'

describe('<CloseButton />', () => {
  it('should renders correct <CloseButton />', () => {
    const onClick = jest.fn()

    const wrapper = shallow(<CloseButton onClick={onClick} />)

    wrapper.find('button').simulate('click')

    expect(onClick).toHaveBeenCalledTimes(1)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has className from prop', () => {
    const className = 'mb-2'
    const onClick = jest.fn()

    const wrapper = shallow(
      <CloseButton onClick={onClick} className={className} />
    )

    expect(wrapper.prop('className')).toContain(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
