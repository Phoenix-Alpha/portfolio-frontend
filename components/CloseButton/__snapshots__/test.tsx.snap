// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton /> should has className from prop 1`] = `
<button
  aria-label="Close"
  className="select-none mb-2"
  onClick={[MockFunction]}
>
  <CloseIcon
    colorHex="#1f2837"
    height={12}
    width={12}
  />
</button>
`;

exports[`<CloseButton /> should renders correct <CloseButton /> 1`] = `
<button
  aria-label="Close"
  className="select-none"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <CloseIcon
    colorHex="#1f2837"
    height={12}
    width={12}
  />
</button>
`;
